alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -222,7 +222,7 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         queryToDocIds: Map<DocLevelQuery, Set<String>>,
         dryrun: Boolean
     ): DocumentLevelTriggerRunResult {
-        val triggerCtx = DocumentLevelTriggerExecutionContext(monitor, trigger)
+        val triggerCtx = DocumentLevelTriggerExecutionContext(monitor, trigger, monitorResult)
```

Comment:
I think instead of the entire ``monitorResult``, we should rather have the ``triggerResults`` set as part of the context for the actions. In DocumentLevelAlerting the two could be different, as trigger results are the final set of document ids sent for alert, post the trigger condition evaluation.

cc: @lezzago

Reply:
Oh that is a good point. I was thinking it to be more in line with the other monitor types and have this as the monitor result instead of the trigger result since that is what the other monitors seem to do.

Reply:
I agree now that we should use trigger results. The other monitor types did not do this because the trigger results from those other monitor types did not have the granularity and useful information.

Reply:
This still needs to address below two concerns - could also be discussed with product for requirements:

1. The monitorResult returned as part of the ctx during the Alerts is the combination of all the documents before the trigger was executed. Since this result ignores the trigger logic and will be repeated across all the Alerts, for the users it could be challenging to map the actual documents which led to those specific Alerts.

2. We also introduced ``related_doc_ids`` as part of the Alerts record, and it is specific to Doc Level Alerts. Since it holds the final result for document Id matching the Alerts, it is more contextual information. However, it introduces an inconsistency for how the users are expected to fetch these compared to other monitor type. 

My proposal would be to deprecate ``related_doc_ids`` in Alerts record slowly, whole have the ``ctx`` hold the documents id specific to those alerts (post trigger execution).